dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/deduplication/DeduplicationControllerMvcTest.java

```diff
@@ -0,0 +1,237 @@
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller.deduplication;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+import java.util.Collections;
+
+import org.hisp.dhis.deduplication.DeduplicationService;
+import org.hisp.dhis.deduplication.DeduplicationStatus;
+import org.hisp.dhis.deduplication.PotentialDuplicate;
+import org.hisp.dhis.fieldfilter.FieldFilterService;
+import org.hisp.dhis.trackedentity.TrackedEntityInstance;
+import org.hisp.dhis.trackedentity.TrackedEntityInstanceService;
+import org.hisp.dhis.trackedentity.TrackerAccessManager;
+import org.hisp.dhis.user.CurrentUserService;
+import org.hisp.dhis.webapi.controller.DeduplicationController;
+import org.hisp.dhis.webapi.controller.exception.ConflictException;
+import org.hisp.dhis.webapi.controller.exception.NotFoundException;
+import org.hisp.dhis.webapi.service.ContextService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Lists;
+
+@RunWith( MockitoJUnitRunner.class )
+public class DeduplicationControllerMvcTest
+{
+    private final static String ENDPOINT = "/" + "potentialDuplicates";
+
+    private MockMvc mockMvc;
+
+    @Mock
+    private DeduplicationService deduplicationService;
+
+    @Mock
+    private TrackedEntityInstanceService trackedEntityInstanceService;
+
+    @Mock
+    private TrackerAccessManager trackerAccessManager;
+
+    @Mock
+    private CurrentUserService currentUserService;
+
+    @Mock
+    private FieldFilterService fieldFilterService;
+
+    @Mock
+    private ContextService contextService;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    private static final String teiA = "trackedentA";
+
+    private static final String teiB = "trackedentB";
+
+    @Before
+    public void setUp()
+    {
+        final DeduplicationController deduplicationController = new DeduplicationController( deduplicationService,
```

Comment:
Just a style suggestion
You could use @InjectMocks and define the deduplicationController as a field